app/src/main/java/com/fastlink/zekrapp/viewModel/ZekirViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package com.fastlink.zekrapp.viewModel
+
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.lifecycle.ViewModel
+import com.fastlink.zekrapp.appData.ZekirSingleton
+import com.fastlink.zekrapp.appData.model.ZekirModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class ZekirViewModel(categoryId: Int) : ViewModel() {
+    val zekirCounter = mutableIntStateOf(0)
+    var zekirs: List<ZekirModel> = emptyList()
```

Comment:
Don't expose mutable variables publicly. You can do this to prevent other classes from modifying your variable:

var zekirs: List<ZekirModel> = emptyList()
                private set